app/assets/javascripts/discourse/app/controllers/composer.js

```diff
@@ -1481,13 +1488,22 @@ export default Controller.extend({
 
   toggleFullscreen() {
     this._saveDraft();
-    if (this.get("model.composeState") === Composer.FULLSCREEN) {
-      this.set("model.composeState", Composer.OPEN);
+
+    const composer = this.model;
+
+    if (composer?.viewFullscreen) {
+      composer?.set("composeState", Composer.OPEN);
     } else {
-      this.set("model.composeState", Composer.FULLSCREEN);
+      composer?.set("composeState", Composer.FULLSCREEN);
+      composer?.toggleProperty("showFullScreenExitPrompt");
```

Comment:
Nitpick, but if this will always set `showFullScreenExitPrompt` to `true`, I prefer being explicit instead of letting `toggleProperty` handle it. It's easier to read and less prone to regression.